cEP-0024.md

```diff
@@ -0,0 +1,252 @@
+IGitt is a possible solution for IGitt implementation. But, it requires extra
+tools to sync GitHub team and GitLab group memberships, which is a manual
+process and isn't automated yet.
+
+Also, team membership informaton is not publicly accessible in GitHub, so any
+system which needs to implement the current assign system would require
+organization authentication tokens.
+
+Best possible solution would be using an increasing difficulty level order,
+so that a user can be assigned to an issue only if user has solved issues of
+lower difficulty level before requesting to get assigned to a higher difficulty
+level issue.
+
+#### Drawbacks
+
+If user completed the lower difficulty level issues in GitHub and want to
+work on a higher difficulty level issue in GitLab, he/she has to repeat the
+newcomer process in GitLab. To resolve this issue, GitMate needs to find a way
+to link GitHub org and GitLab org, this would be easy when they have same name,
+as in the case of `coala`. But, resolving this issue is out of scope for this
+project.
+
+If a user is worthy enough to work on the higher level difficulty issues
+directly, then that user needs to be manually assigned by the maintainers until
+that user completes at least one low difficulty issue. Even with current assign
+system, manual intervention by maintainers is needed to add user to developers
+team.
+
+## New Features
+
+### Auto-assign issue when it is mentioned in any of PR commits
+
+When a PR is opened or updated, if any of PR's commit message referenced
+any issues with keywords `Fixes` or `Closes` and there is no
+CI failure in PR, then auto assign PR author to the mentioned unassigned
+issues.
+
+If someone else is working on the mentioned issue, comment will be added
+on the PR (configurable).
+
+If PR author is not a member of the organization, then he/she can't be assigned
+to any of the issues, a label(configurable) will be added on the PR so that
+maintainers can identify these type of PRs and a comment will be added on PR
+informing PR author to get invited to the organization before working on any
+issue.
+
+coala can configure this plugin as:
+
+**Comment message when someone else is assigned to mentioned issues(use
+`{issues}` as placeholder to get links of all assigned issues):**
+`Sorry, but the issues {issues} are assigned to someone else. Please try working
+on some other issues.`
+
+**Label to represent that PR author is not a member of the org:**
+`author/non-member`
+
+### Auto-reject PR opened on certain branch names
+
+Normally, contributors should make changes in different branch from master
+branch of the forked repository, and submit a PR. But, often coala receives PRs
+from master branch of the forked repository.
+
+coala needs mechanism such that when a PR is opened or updated, GitMate
+automatically closes PRs opened on certain branch names(master), adds certain
+label to the PR so that maintainers can identify these type of PRs and comments
```

Comment:
When the bot is already commenting on the PR and informing to work on a different branch than master. There is no need for reviewers to comment on the PR themselves.

Reply:
Yeah, there is no need for reviewers to comment on the PR, I just said a label will be added so that maintainers can identify these type of PRs